docs/user-guide.md

```diff
@@ -328,6 +328,15 @@ performed.
 *   `HANDLE_EA_EL3_FIRST`: When defined External Aborts and SError Interrupts
     will be always trapped in EL3 i.e. in BL31 at runtime.
 
+*   `HW_ASSISTED_COHERENCY`: On most ARM systems to-date, platform-specific
+    software operations are required for CPUs to enter and exit coherency.
+    However, there exists newer systems where CPUs' entry to and exit from
+    coherency is managed in hardware. Such systems require software to only
```

Comment:
(I had left a comment earlier, but it seemed that I started a review in Github, rather than publishing the comment itself. So I'm not sure if my earlier reply was publicly visible; I re-compose my earlier reply).

Without being specific, this only implies that, as soon a CPU is out of reset and enter PSCI state coordination, it's coherent with all other CPUs in the system, and is "ready to go".

Typically in multi-cluster ARM systems, the clusters are connected via. CCI or similar IP to keep all clusters coherent. This means that, before CPUs in those clusters can be assumed to be coherent, the CCI needs programming. Because of this, TF presently delays assuming (a sub-optimal but necessary step to ensure correctness) them to be coherent until the IP is programmed.

However, with this build option, TF is told that all CPUs in the system are coherent with each other as soon as they're out of reset, without needing programming of any IP. I.e. regardless of which cluster they're in, they all _effectively_ belong to one and same coherency domain.